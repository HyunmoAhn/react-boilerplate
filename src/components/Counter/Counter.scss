.Counter {
  border: 5px solid black;
  margin: auto;
  width: 50%;
}

.Counter__value {
  margin: 15px;
  width: 50px;
}

.Counter__btn {
  width: auto;
  height: auto;
}