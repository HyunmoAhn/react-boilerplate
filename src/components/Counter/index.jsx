import React from 'react';

class Counter extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			value: 0,
		};
		this.handleIncrement = this.handleIncrement.bind(this);
		this.handleDecrement = this.handleDecrement.bind(this);
	}

	handleIncrement() {
		this.setState({ value: this.state.value + 1 });
	}

	handleDecrement() {
		this.setState({ value: this.state.value - 1 });
	}

	render() {
		return (
			<div>
				{this.state.value}
				<input type="button" value="increment" onClick={this.handleIncrement} />
				<input type="button" value="decrement" onClick={this.handleDecrement} />
				{this.props.children}
			</div>
		)
	}
};

export default Counter;