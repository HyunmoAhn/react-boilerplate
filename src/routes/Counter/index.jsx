import CounterContainer from './containers/CounterContainer';
import Children from './routes/children';

export default () => ({
	path: 'counter',
	component: CounterContainer,
	childRoutes: [
		Children(),
	],
});