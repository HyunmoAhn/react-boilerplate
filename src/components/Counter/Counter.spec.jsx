import React from 'react';
import { shallow } from 'enzyme';

import Counter from './';

describe('<Counter />', () => {
	it('should match to snapshot when render default', () => {
		const wrapper = shallow(<Counter />);

		expect(wrapper).toMatchSnapshot();
	});
});