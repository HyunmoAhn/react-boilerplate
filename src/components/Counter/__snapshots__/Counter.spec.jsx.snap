// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should match to snapshot when render default 1`] = `
<div
  className="Counter"
>
  <span
    className="Counter__value"
  >
    0
  </span>
  <input
    className="Counter__btn"
    onClick={[Function]}
    type="button"
    value="increment"
  />
  <input
    className="Counter__btn"
    onClick={[Function]}
    type="button"
    value="decrement"
  />
  <input
    className="Counter__btn"
    onClick={[Function]}
    type="button"
    value="Open ALERT Modal"
  />
        
  <input
    className="Counter__btn"
    onClick={[Function]}
    type="button"
    value="Open CONFIRM Modal"
  />
</div>
`;
